test/Common/DurableTaskLifeCycleNotificationTest.cs

```diff
@@ -1342,6 +1342,41 @@ public void OrchestrationCustomHelperTypeFallback()
             Assert.IsType<NullLifeCycleNotificationHelper>(lifeCycleNotificationHelper);
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task OrchestrationCustomHelperTypeDependencyInjection()

```

Comment:
This test doesn't quite test that Dependency Injection is happening, as the `MockLifeCycleNotificationHandler` is being handled by the test's dependency injection, not our production dependency injection.

That being said, this test is well structured to test the business logic around our `ILifeCycleNotificationHelper` usage. Instead of using a counter though, I would make a `List<string>` of all of the operations called, so that we can make sure that the order of notification calls (i.e. OrchestratorStarted then OrchestratorCompleted) is correct.

Reply:
If the test is run for v2, it should be resolved from DI. (v1 cannot use DI, it is dependency injection for testing as you indicated.)

This test case is to check if `MockLifeCycleNotificationHelper` is resolved from DI and called, it doesn't consider the calling order.

Reply:
I see. This test is looking for customer facing dependency injection (for test purposes etc), not that our production code path is properly populating the correct `ILifeCycleNotificationHelper`.  In that case, I think this test is sufficient as is.